From c576a66fe8a3d3b26fe2529f4b4591f54c8fc72b Mon Sep 17 00:00:00 2001
From: Andrew LaFrance <robert.a.lafrance5.civ@us.navy.mil>
Date: Mon, 21 Apr 2025 23:10:02 +0000
Subject: [PATCH 24/29] simCore: Add GPKG extension to GDAL Image file lists

**JIRA Issue:** SIM-7877

**Description:** Add GeoPackage extension (*.gpkg) to the list of GDAL-supported imagery files.

**Notes:** GDAL has supported GPKG since mid-2022, but looks like the extension was never added to the list, so I had to switch to "All files" to see and load the GeoPackage maps I was loading.

**Testing Performed:** Manual in Release, loaded GPKG files through the "Quick Add Imagery" and "Add Image Layer" menus in the Map editor (reaching both of the strings edited here). No issues, imagery loaded as expected
---
 SDK/simCore/String/FilePatterns.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/SDK/simCore/String/FilePatterns.h b/SDK/simCore/String/FilePatterns.h
index 9e6f06ef..00a97052 100644
--- a/SDK/simCore/String/FilePatterns.h
+++ b/SDK/simCore/String/FilePatterns.h
@@ -322,7 +322,7 @@ static const std::string ALL_DATA_FILE_PATTERNS = ALL_SIMDIS_FILE_PATTERNS +
 
 /** GDAL files for osgEarth GDAL driver image and elevation layers */
 static const std::string GDAL_IMAGE_LAYER_FILE_PATTERNS =
-  "GDAL Files (*.tif,*.tiff,*.jpeg,*.png,*.img,*.sid,*.jp2,*.bil,*.hgt)\n"
+  "GDAL Files (*.tif,*.tiff,*.jpeg,*.png,*.img,*.sid,*.jp2,*.bil,*.hgt,*.gpkg)\n"
   "All Files (*)";
 
 /** MBTiles files for osgEarth MBTiles driver image layers */
@@ -337,10 +337,10 @@ static const std::string MAPBOX_GL_IMAGE_LAYER_FILE_PATTERNS =
 
 /** Pattern for GDAL, DB, MBTiles, MapBox GL, other files -- image file layer pattern */
 static const std::string FILEBASED_IMAGE_LAYER_FILE_PATTERNS =
-  "All Layer Files (*.mbtiles,*.db,*.tif,*.tiff,*.jpeg,*.png,*.img,*.sid,*.jp2,*.bil,*.json,Conf.xml,conf.xml)\n"
+  "All Layer Files (*.mbtiles,*.db,*.tif,*.tiff,*.jpeg,*.png,*.img,*.sid,*.jp2,*.bil,*.gpkg,*.json,Conf.xml,conf.xml)\n"
   "MBTiles Files (*.mbtiles)\n"
   "DB Files (*.db)\n"
-  "GDAL Files (*.tif,*.tiff,*.jpeg,*.png,*.img,*.sid,*.jp2,*.bil)\n"
+  "GDAL Files (*.tif,*.tiff,*.jpeg,*.png,*.img,*.sid,*.jp2,*.bil,*.gpkg)\n"
   "MapBox GL Image Style Files (*.json)\n"
   "Tile Package Configuration (Conf.xml,conf.xml)\n"
   "All Files (*)";
-- 
2.47.1.windows.2

